readthedocs/doc_builder/director.py

```diff
@@ -335,6 +340,185 @@ def run_build_job(self, job):
         for command in commands:
             environment.run(*command.split(), escape_command=False, cwd=cwd)
 
+    def run_build_commands(self):
+        reshim_commands = (
+            ("pip", "install"),
+            ("conda", "create"),
+            ("conda", "install"),
+            ("mamba", "create"),
+            ("mamba", "install"),
+            ("poetry", "install"),
+        )
+        cwd = self.data.project.checkout_path(self.data.version.slug)
+        environment = self.vcs_environment
+        for command in self.data.config.build.commands:
+            environment.run(*command.split(), escape_command=False, cwd=cwd)
+
+            # Execute ``asdf reshim python`` if the user is installing a
+            # package since the package may contain an executable
+            for reshim_command in reshim_commands:
+                # Convert tuple/list into set to check reshim command is a
+                # subset of the command itself. This is to find ``pip install``
+                # but also ``pip -v install`` and ``python -m pip install``
+                if set(reshim_command).issubset(command.split()):
```

Comment:
you could declare `reshim_commands` as sets, so you don't have to convert it to a set every time.